fuzz/fuzz_targets/smt.rs

```diff
@@ -0,0 +1,80 @@
+#![no_main]
+
+use libfuzzer_sys::fuzz_target;
+use miden_crypto::{merkle::Smt, hash::rpo::RpoDigest, Word, Felt, ONE};
+use rand::Rng; // Needed for randomizing the split percentage
+
+struct FuzzInput {
+    entries: Vec<(RpoDigest, Word)>,
+    updates: Vec<(RpoDigest, Word)>,
+}
+
+impl FuzzInput {
+    fn from_bytes(data: &[u8]) -> Self {
+        let mut rng = rand::thread_rng();
+        let split_percentage = rng.gen_range(20..80); // Randomly choose between 20% and 80%
+
+        let split_index = (data.len() * split_percentage) / 100;
+        let (construction_data, update_data) = data.split_at(split_index);
+
+        let entries = Self::parse_entries(construction_data);
+        let updates = Self::parse_entries(update_data);
+
+        Self { entries, updates }
+    }
+
+    fn parse_entries(data: &[u8]) -> Vec<(RpoDigest, Word)> {
+        let mut entries = Vec::new();
+        let num_entries = data.len() / 40; // Each entry is 40 bytes
+
+        for chunk in data.chunks_exact(40).take(num_entries) {
+            let key = RpoDigest::new([
+                Felt::new(u64::from_le_bytes(chunk[0..8].try_into().unwrap())),
+                Felt::new(u64::from_le_bytes(chunk[8..16].try_into().unwrap())),
+                Felt::new(u64::from_le_bytes(chunk[16..24].try_into().unwrap())),
+                Felt::new(u64::from_le_bytes(chunk[24..32].try_into().unwrap())),
+            ]);
+            let value = [
+                ONE,
+                ONE,
+                ONE,
+                Felt::new(u64::from_le_bytes(chunk[32..40].try_into().unwrap())),
+            ];
+            entries.push((key, value));
+        }
+
+        entries
+    }
+}
+
+fuzz_target!(|data: &[u8]| {
+    let fuzz_input = FuzzInput::from_bytes(data);
+    run_fuzz_smt(fuzz_input);
+});
+
+fn run_fuzz_smt(fuzz_input: FuzzInput) {
+    let sequential_result = Smt::with_entries(fuzz_input.entries.clone());
+    let parallel_result = Smt::with_entries(fuzz_input.entries);
```

Comment:
I assume the first is intended to be `Smt::fuzz_with_entries_sequential`, otherwise it's using the concurrent implementation for both.